config/config.go

```diff
@@ -1312,6 +1364,8 @@ type RemoteWriteConfig struct {
 	// values arbitrarily into the overflow maps of further-down types.
 	HTTPClientConfig HTTPClientConfig `yaml:",inline"`
 
+	RemoteQueueConfig RemoteQueueConfig `yaml:",inline"`
```

Comment:
I wonder if it would be better to nest these config options under a key like `queue_config` instead of inlining even more structs. Less pollution of the main remote write namespace and thus hopefully easier to read the docs about it? Arguably we could also do that with the HTTP client options everywhere if it wouldn't be a breaking change.